internal/pkg/runtime/engine/imgbuild/create_linux.go

```diff
@@ -135,36 +148,48 @@ func (e *EngineOperations) CreateContainer(ctx context.Context, pid int, rpcConn
 		}
 	}
 
-	dest = filepath.Join(sessionPath, "dev")
+	dest = filepath.Join(sessionRootFs, "dev")
 	sylog.Debugf("Mounting /dev at %s\n", dest)
 	if err := rpcOps.Mount("/dev", dest, "", syscall.MS_BIND|syscall.MS_REC, ""); err != nil {
 		return fmt.Errorf("mount /dev failed: %s", err)
 	}
 
-	dest = filepath.Join(sessionPath, "etc", "resolv.conf")
-	sylog.Debugf("Mounting /etc/resolv.conf at %s\n", dest)
-	if err := rpcOps.Mount("/etc/resolv.conf", dest, "", flags, ""); err != nil {
-		return fmt.Errorf("mount /etc/resolv.conf failed: %s", err)
+	// copy /etc/resolv.conf to tmpfs and bind copy into container
+	sessionResolv, err := stageFile("/etc/resolv.conf", sessionPath)
+	if err != nil {
+		return err
+	}
+
+	dest = filepath.Join(sessionRootFs, "etc", "resolv.conf")
+	sylog.Debugf("Mounting %s at %s\n", sessionResolv, dest)
+	if err := rpcOps.Mount(sessionResolv, dest, "", flags, ""); err != nil {
+		return fmt.Errorf("mount %s failed: %s", sessionResolv, err)
 	}
 	if !insideUserNs {
 		if err := rpcOps.Mount("", dest, "", syscall.MS_REMOUNT|flags, ""); err != nil {
 			return fmt.Errorf("remount /etc/resolv.conf failed: %s", err)
 		}
 	}
 
-	dest = filepath.Join(sessionPath, "etc", "hosts")
-	sylog.Debugf("Mounting /etc/hosts at %s\n", dest)
-	if err := rpcOps.Mount("/etc/hosts", dest, "", flags, ""); err != nil {
-		return fmt.Errorf("mount /etc/hosts failed: %s", err)
+	// copy /etc/hosts to tmpfs and bind copy into container
+	sessionHosts, err := stageFile("/etc/hosts", sessionPath)
+	if err != nil {
+		return err
+	}
+
+	dest = filepath.Join(sessionRootFs, "etc", "hosts")
+	sylog.Debugf("Mounting %s at %s\n", sessionHosts, dest)
+	if err := rpcOps.Mount(sessionHosts, dest, "", flags, ""); err != nil {
```

Comment:
flags here has `MS.RD_ONLY` given that rpm is still trying to modify our staged copy, can we just mount it `rw`? Any changes made are only relevant for the session?

Reply:
Oops, fixed